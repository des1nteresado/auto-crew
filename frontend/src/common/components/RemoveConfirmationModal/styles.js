const styles = (theme) => ({
  closeButton: {
    color: theme.palette.black,
    minWidth: 20,
    maxWidth: 20,
    height: 40,
    position: 'absolute',
    top: 11,
    right: 15,
    padding: 0,
  },
  modalTitle: {
    color: theme.palette.black,
    marginBottom: 20,
    padding: 10,
  },
  cancelWrapper: {
    marginBottom: 30,
  },
  buttonsWrapper: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  controlButton: {
    marginRight: 50,
    color: theme.palette.black,

    '&:last-of-type': {
      marginRight: 0,
    },
  },
});

export default styles;
