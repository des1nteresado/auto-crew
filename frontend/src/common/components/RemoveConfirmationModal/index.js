import React from 'react';
import PropTypes from 'prop-types';
import { Box, Typography, withStyles } from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';

import CommonButton from '../CommonButton';
import CommonDialog from '../CommonDialog';

import styles from './styles';

const RemoveConfirmationModal = ({
  classes,
  handleSubmit,
  subTitle,
  handleRemoveModalClose,
  isOpen,
}) => {
  return (
    <CommonDialog isOpen={isOpen} onClose={handleRemoveModalClose} style={classes.commonDialog}>
      <Box className={classes.cancelWrapper}>
        <CommonButton
          className={classes.closeButton}
          startIcon={<CloseIcon />}
          label=""
          variant="text"
          size="large"
          onClick={handleRemoveModalClose}
        />
      </Box>
      <Typography variant="h3" align="center" className={classes.modalTitle}>
        Are you sure, that you want to delete this {subTitle}?
      </Typography>
      <Box className={classes.buttonsWrapper}>
        <CommonButton
          className={classes.controlButton}
          label="Yes"
          variant="contained"
          onClick={handleSubmit}
        />
        <CommonButton
          className={classes.controlButton}
          label="No"
          variant="outlined"
          onClick={handleRemoveModalClose}
        />
      </Box>
    </CommonDialog>
  );
};

RemoveConfirmationModal.propTypes = {
  classes: PropTypes.oneOfType([PropTypes.object]).isRequired,
  isOpen: PropTypes.bool.isRequired,
  subTitle: PropTypes.string.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleRemoveModalClose: PropTypes.func.isRequired,
};

export default React.memo(withStyles(styles)(RemoveConfirmationModal));
