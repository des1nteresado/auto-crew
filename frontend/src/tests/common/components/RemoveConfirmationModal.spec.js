import React from 'react';
import { shallow } from 'enzyme';

import RemoveConfirmationModal from 'common/components/RemoveConfirmationModal';

const defaultProps = {
  isOpen: true,
  subTitle: 'title',
  handleSubmit: jest.fn(),
  handleRemoveModalClose: jest.fn(),
};

const setUp = (props) => shallow(<RemoveConfirmationModal {...defaultProps} {...props} />).dive();

describe('RemoveConfirmationModal', () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });

  it('renders RemoveConfirmationModal component correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
