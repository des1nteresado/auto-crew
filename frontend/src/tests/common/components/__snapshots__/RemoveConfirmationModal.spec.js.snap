// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveConfirmationModal renders RemoveConfirmationModal component correctly 1`] = `
<Memo(WithStyles(CommonDialog))
  isOpen={true}
  onClose={[MockFunction]}
>
  <Styled(MuiBox)
    className="RemoveConfirmationModal-cancelWrapper-3"
  >
    <Memo(CommonButton)
      className="RemoveConfirmationModal-closeButton-1"
      label=""
      onClick={[MockFunction]}
      size="large"
      startIcon={<UNDEFINED />}
      variant="text"
    />
  </Styled(MuiBox)>
  <WithStyles(ForwardRef(Typography))
    align="center"
    className="RemoveConfirmationModal-modalTitle-2"
    variant="h3"
  >
    Are you sure, that you want to delete this 
    title
    ?
  </WithStyles(ForwardRef(Typography))>
  <Styled(MuiBox)
    className="RemoveConfirmationModal-buttonsWrapper-4"
  >
    <Memo(CommonButton)
      className="RemoveConfirmationModal-controlButton-5"
      label="Yes"
      onClick={[MockFunction]}
      variant="contained"
    />
    <Memo(CommonButton)
      className="RemoveConfirmationModal-controlButton-5"
      label="No"
      onClick={[MockFunction]}
      variant="outlined"
    />
  </Styled(MuiBox)>
</Memo(WithStyles(CommonDialog))>
`;
